benchmarks/huffman_coding/oop_c#/Program.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System.Text;
+using System.Collections.Generic;
+using System;
+using System.IO;
+
+namespace oop_c_
+{
+    class Program
+    {
+        static string TEST_STRING = File.ReadAllText("benchmarks/huffman_coding/lines.txt");
+
+        static void Main(string[] args)
+        {
+            Huffman huffman = new Huffman(TEST_STRING);
+            string encodedString = huffman.Encode(TEST_STRING);
+            System.Console.WriteLine(encodedString.Length);
+        }
+    }
+
+    public abstract class HuffmanTree : IComparable<HuffmanTree>
+    {
+        public int Id { get; set; } // Nessecary to ensure total order
+        public int Frequency { get; }
+        public HuffmanTree(int freq, int id) =>
+            (Frequency, Id) = (freq, id);
+
+        // Sorts based on lowest frequency
+        public int CompareTo(HuffmanTree obj)
+        {
+            int res = Frequency - obj.Frequency;
+            return res == 0 ? Id.CompareTo(obj.Id) : res;
+        }
+    }
+
+    public class HuffmanLeaf : HuffmanTree
+    {
+        public char Character { get; } // the character this leaf represents
+        public HuffmanLeaf(char c, int freq, int id) : base(freq, id) =>
+            Character = c;
+    }
+
+    public class HuffmanNode : HuffmanTree
+    {
+        public HuffmanTree Left, Right; //subtrees
+        public HuffmanNode(HuffmanTree left, HuffmanTree right, int id) : 
+            base(left.Frequency + right.Frequency, id) =>
+                (Left,Right) = (left,right);
+    }
+
+    public class Huffman
+    {
+        Dictionary<char, string> SymbolTable { get; set; }
+        public Huffman(string stringToEncode)
+        {
+            SymbolTable = new Dictionary<char, string>();
+            var frequencies = new Dictionary<char, int>();
+            // read each character and record the frequencies
+            foreach (char c in stringToEncode)
+            {
+                if (!frequencies.ContainsKey(c))
+                    frequencies[c] = 0;
+                frequencies[c]++;

```

Comment:
```suggestion
                if (!frequencies.ContainsKey(c))
                    frequencies[c] = 1;
                else
                    frequencies[c]++;

```
Kan godt ske at dette er lidt ligemeget